packer/resources/features/ssh-keys/install.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+set -e
+
+function HELP {
+>&2 cat << EOF
+
+  Usage: ${0} -k key-s3-url [-u ubuntu]
+
+  This script retrieves public SSH keys from AWS S3 and overwrites
+  ~/.ssh/authorized_keys for the defined user. 
+
+    -k key-s3-url The full S3 URL of the public SSH keys (e.g. s3://mybucket/ssh-keys.txt).
+
+    -u user       [optional] the user to install the SSH keys for. Defaults to ubuntu.
+
+    -h            Displays this help message. No further functions are
+                  performed.
+
+EOF
+exit 1
+}
+
+# Process options
+while getopts k:u:h FLAG; do
+  case $FLAG in
+    k)
+      S3_KEY_URL=$OPTARG
+      ;;
+    u)
+      SSH_USER=$OPTARG
+      ;;
+    h)  #show help
+      HELP
+      ;;
+  esac
+done
+shift $((OPTIND-1))
+
+if [ -z "${S3_KEY_URL}" ]; then
+  echo "Must specify an S3 URL"
+  exit 1
+fi
+
+if [ -z "${SSH_USER}" ]; then
+  SSH_USER="ubuntu"
+fi
+
+TEMPORARY_SSH_KEY_FILE="/tmp/authorized_keys.$$"
```

Comment:
What's the reason for the temp file? Why not download directly to `~/.ssh/authorized_keys`?